client/beefy/rpc/src/lib.rs

```diff
@@ -393,4 +410,74 @@ mod tests {
 		assert_eq!(recv_sub_id, sub_id);
 		assert_eq!(recv_commitment, commitment);
 	}
+
+	#[test]
+	fn should_not_send_signed_commitment_multiple_times() {
+		let (sender, stream) = BeefyBestBlockStream::<Block>::channel();
+		let (io, commitment_sender) = setup_io_handler_with_best_block_stream(stream);
+		let (meta, receiver) = setup_session();
+
+		// Subscribe
+		let sub_request =
+			r#"{"jsonrpc":"2.0","method":"beefy_subscribeJustifications","params":[],"id":1}"#;
+		let resp = io.handle_request_sync(sub_request, meta.clone());
+		let resp: Output = serde_json::from_str(&resp.unwrap()).unwrap();
+
+		let sub_id = match resp {
+			Output::Success(success) => success.result,
+			_ => panic!(),
+		};
+
+		// Notify with commitment
+		let commitment = create_commitment(5);
+		let r: Result<(), ()> = commitment_sender.notify(|| Ok(commitment.clone()));
+		r.unwrap();
+		let hash = BlakeTwo256::hash(b"43");
+		let r: Result<(), ()> = sender.notify(|| Ok((hash, 5)));
+		r.unwrap();
+
+		// This commitment should be filtered out
+		let r: Result<(), ()> = commitment_sender.notify(|| Ok(commitment.clone()));
+		r.unwrap();
+
+		let commitment_1 = create_commitment(6);
+
+		let r: Result<(), ()> = commitment_sender.notify(|| Ok(commitment_1.clone()));
+		r.unwrap();
+
+		// Inspect what we received
+		// We should have received only two commitments
+		let recvs = futures::executor::block_on(receiver.take(2).collect::<Vec<_>>());
+		let recv: Notification = serde_json::from_str(&recvs[0]).unwrap();
+		let mut json_map = match recv.params {
+			Params::Map(json_map) => json_map,
+			_ => panic!(),
+		};
```

Comment:
Simply compare to a static string the expected response

Reply:
The response string contains a subscription id that is dynamically generated on each run, so the static string might not work.

Reply:
> The response string contains a subscription id that is dynamically generated on each run, so the static string might not work.

no way to mock subscription id?